packages/upload-api/src/space.js

```diff
@@ -0,0 +1,48 @@
+import { Space } from '@web3-storage/capabilities'
+import { DID } from '@ucanto/validator'
+import * as Provider from '@ucanto/server'
+import * as API from './types.js'
+
+/**
+ * @param {API.Input<Space.info>} input
+ * @param {API.SpaceServiceContext} ctx
+ * @returns {Promise<API.Result<{ did: API.SpaceDID }, API.Failure>>}
+ */
+export const info = async ({ capability }, ctx) => {
+  const { provisionsStorage: provisions } = ctx
+
+  const spaceDid = capability.with
+  if (!DID.match({ method: 'key' }).is(spaceDid)) {
+    /** @type {API.SpaceUnknown} */
+    const unexpectedSpaceDidFailure = {
+      name: 'SpaceUnknown',
+      message: `can only get info for did:key spaces`,
+    }
+    return {
+      error: unexpectedSpaceDidFailure,
+    }
+  }
+
+  const result = await provisions.hasStorageProvider(spaceDid)
```

Comment:
As in all other cases we should consider provider in this case it would be `invocation.audience`

Reply:
ok I have added an issue to the repo to track this and the others - would be great to get this in soon https://github.com/web3-storage/w3up/issues/802